src/client/service.ts

```diff
@@ -149,17 +149,19 @@ export class Service {
     // lowercase first letter
     let methodName = methodMeta.name.charAt(0).toLowerCase() + methodMeta.name.slice(1);
     this.handle[methodName] = (argument?: any,
+                               metadata?: any,
                                callback?: (error?: any, response?: any) => void) => {
-      if (typeof argument === 'function' && !callback) {
+      if (typeof argument === 'function' && !callback && !metadata) {
```

Comment:
IIUC, this works to only pass a callback, but as soon as an argument is provided, then the metadata param becomes required, right? This should probably handle the case of passing a callback with arg, but no metadata